### ISSUE TITLE
BDA3 Chapter 3 discussion of Effective sample size is flawed (needs update)

### ISSUE THREAD

**Issue by rpgoldman** [pymc-devs/resources#78]
I was running through the notebook for Chapter 3, and examining the "Example. Pre-election polling" section.  This uses a no longer available PyMC3 API function to compute `n_eff`.  I tried changing this to use the corresponding arviz function `az.stats.ess`, and got the anomalous results in the following screen-grab.  Specifically, the `pm.summary()` measures of ESS and the `az.stats.ess()` measures don't agree:

![image](https://user-images.githubusercontent.com/3274/79081017-30699b00-7cdf-11ea-9c4d-d575591eb195.png)

Can anyone explain why the ESS for θ1, θ2, and θ3 are so different in the two measures?  Also, shouldn't we remove the mention of `n_eff`, which appears nowhere in the output of _either_ of the two ESS measures.

Note that I am _assuming_ here that "effective sample size" and "number of effective samples" are the same.

**Comment by aloctavodia** [comment]
Hi @rpgoldman, notice that the default method for `az.ess` is `bulk` and those values agree (within the rounding error) with the ones reported by `pm.summary` in your example. 

I agree we should update the examples and we should also directly use arviz functions when possible.

**Comment by rpgoldman** [comment]
@aloctavodia Hi.  I just updated my issue report as I discovered that the ESS was "bulk," not "mean." I also tried to more clearly state what I think is the problem and what I think could be done to improve matters.

A person who has only gotten to chapter 3 of BDA is likely not to understand how sampling should be diagnosed...

**Comment by rpgoldman** [comment]
Also note that we *must* use the arviz function here -- the PyMC3 one no longer exists (or it has been renamed).

**Comment by rpgoldman** [comment]
@aloctavodia BTW, this is leading me to patch the Arviz docs for the `ess` function.  One thing that is odd in there is that it describes the `var_names` as

> Names of variables to include in the effective_sample_size_mean report

I think this should just be "names of variables for which to compute ESS", right? I'm betting the `effective_sample_size_mean` is obsolete, since we offer multiple methods, not just mean.

**Comment by aloctavodia** [comment]
@rpgoldman I totally agree with the statement "A person who has only gotten to chapter 3 of BDA is likely not to understand how sampling should be diagnosed..." But I think the main purpose of these "resources" is to provide a port of the code used in BDA and other books to PyMC3/4/ArviZ, not sure we should "complement" the content of those books here. Maybe we can add a few lines saying something about ess  but a more detailed explanation belong somewhere else maybe we can add a link to this material https://github.com/arviz-devs/Exploratory-Analysis-of-Bayesian-Models/tree/master/content (that is far from ready) but it's main intention is precisely to explain the "theory"  and "good practices" around methods implemented in ArviZ.

Right, the value returned by ESS is "bulk,"  "mean" is the value of the old way of computing ess. Probably we should revisit winch values we show in summary, as this can be certainly confusing and also address the issue with the docstring you mention. This all are important issues and we should addressed. 

Are you willing to send a PR to fix this issue?

**Comment by rpgoldman** [comment]
Sure, I'll try a PR, but it will *definitely* need a review!

**Comment by rpgoldman** [comment]
@aloctavodia If you get a chance, please see my Arviz docstring fix (I don't have permissions to assign it to you for review).

**Comment by aloctavodia** [comment]
Sure, thanks! I will do.
